python/lsst/pipe/tasks/assembleCoadd.py

```diff
@@ -1362,3 +1374,327 @@ def detectClipBig(self, tempExpClipList, clipFootprints, clipIndices, maskClipVa
+                                                        maxNumEpochs=maxNumEpochs)
+            dilatedSpanSetList = [s.dilated(self.config.growMaskBy, afwGeom.Stencil.CIRCLE)
+                                  for s in filteredSpanSetList]
+            spanSetArtifactList.append(dilatedSpanSetList)
+            spanSetNoDataMaskList.append(spanSetNoDataMask)
+        return pipeBase.Struct(artifacts=spanSetArtifactList,
+                               noData=spanSetNoDataMaskList)
+
+    def computeAltMaskList(self, tempExpRefList, maskSpanSets):
+        """!
+        \brief Apply artifact span set lists to masks
+
+        @param tempExpRefList: List of data references to warps
+        @param imageScalerList: List of image scalers
+
+        return List of alternative masks
+
+        Add artifact span set list as "CLIPPED" plane and NaNs to existing "NO_DATA" plane
+        """
+        spanSetMaskList = maskSpanSets.artifacts
+        spanSetNoDataList = maskSpanSets.noData
+        altMaskList = []
+        for warpRef, artifacts, noData in zip(tempExpRefList, spanSetMaskList, spanSetNoDataList):
+            warp = warpRef.get(self.getTempExpDatasetName(self.config.warpType), immediate=True)
+            mask = warp.maskedImage.mask
+            maskClipValue = mask.addMaskPlane("CLIPPED")
+            noDataValue = mask.addMaskPlane("NO_DATA")
+            _ = [s.clippedTo(mask.getBBox()).setMask(mask, 2**maskClipValue) for s in artifacts]
+            _ = [s.clippedTo(mask.getBBox()).setMask(mask, 2**noDataValue) for s in noData]
+            altMaskList.append(mask)
+        return altMaskList
+
+    def _filterArtifacts(self, spanSetList, epochCountImage, maxNumEpochs=None, minPixels=None):
+        if minPixels is None:
+            minPixels = self.config.minPixels
+        maskSpanSetList = []
+        x0, y0 = epochCountImage.getXY0()
+        for i, span in enumerate(spanSetList):
+            y, x = span.indices()
+            if len(y) < self.config.minPixels:
+                continue
+            counts = epochCountImage.array[[y1 - y0 for y1 in y], [x1 - x0 for x1 in x]]
+            idx = numpy.where((counts > 0) & (counts <= maxNumEpochs))
+            percentBelowThreshold = len(idx[0]) / len(counts)
+            if percentBelowThreshold > self.config.spatialThreshold:
+                maskSpanSetList.append(span)
+        return maskSpanSetList
+
+    def _snrToBinaryArr(self, arrIn):
+        with numpy.errstate(invalid='ignore'):
+            arr = numpy.where(arrIn >= self.config.chiThreshold, 1, 0)
+            if self.config.doMaskNegative:
+                arr[numpy.where(arrIn <= -self.config.chiThreshold)] = 1
+        return arr
+
+    def _snrToBinaryIm(self, im):
+        arr = self._snrToBinaryArr(im.array)
+        return afwImage.ImageU(arr.astype(numpy.uint16), xy0=im.getXY0())
+
+    def _makeChiIm(self, maskedImage):
+        chiIm = maskedImage.image.Factory(1./numpy.sqrt(maskedImage.variance.array))
```

Comment:
Do you need to worry about invalid variance values being present (<=0 or nan)?

Reply:
No, that seems fine.

Reply:
I think that that's standard IEEE754 and can be relied upon.